### PR TITLE
Fix/basket

### PR DESCRIPTION
**Pull request by oleksandr-bratashchuk**

### CODE REVIEW

File: src/api/products.ts [hunk 1]
```diff
@@ -24,3 +24,33 @@ export const fetchProducts = async (
...
+  await axios.put(`${BASE_URL}/${updatedProduct.id}`);
+
+  return updatedProduct;
+};
+
+export const removeProduct = async (
+  product: Product
+): Promise<Product | string> => {
+  const { data } = await axios.delete(`${BASE_URL}/${product.id}`);
+  console.log(data);
```
Comment (oleksandr-bratashchuk) [inline src/api/products.ts:54]: @oleksandr-bratashchuk remove console.log()

File: src/components/ProductForm/ProductForm.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+export const ProductForm: React.FC<ProductFormProps> = ({
+  submitCallback,
+  product,
+  setModalOpen,
+  setShowNotification,
+  type,
+}) => {
+  const [categories, setCategories] = React.useState<string[]>([]);
+
+  React.useEffect(() => {
```
Comment (Bohdan-ya) [inline src/components/ProductForm/ProductForm.tsx:42]: @oleksandr-bratashchuk use useMemo instead useState and useEffect

File: src/components/ProductForm/ProductForm.tsx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    fetchCategories().then(setCategories);
+  }, []);
+
+  return (
+    <Formik
+      initialValues={
```
Comment (Bohdan-ya) [inline src/components/ProductForm/ProductForm.tsx:48]: @oleksandr-bratashchuk add separate object for this , and use it instead

File: src/components/ProductForm/ProductForm.tsx [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        product || {
+          title: '',
+          description: '',
+          price: '',
+          category: '',
+          images: [''],
+        }
+      }
+      validationSchema={Yup.object().shape({
```
Comment (Bohdan-ya) [inline src/components/ProductForm/ProductForm.tsx:57]: @oleksandr-bratashchuk the same

File: src/components/ProductForm/ProductForm.tsx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        title: Yup.string().required('Title is required'),
+        price: Yup.number().min(1, 'invalid price').required('Price is required'),
+        category: Yup.string().required('choose a category'),
+      })}
+      isInitialValid={type === 'Edit'}
+      onSubmit={async (values, { setSubmitting }) => {
```
Comment (Bohdan-ya) [inline src/components/ProductForm/ProductForm.tsx:63]: @oleksandr-bratashchuk the same. Add function for this

File: src/contexts/SearchContext.tsx [hunk 6]
```diff
@@ -1,35 +1,36 @@
...
+    query: '',
+  });
 
 export const SearchProvider: React.FC<React.PropsWithChildren<{}>> = ({
   children,
 }) => {
-  const [categorie, setCategorie] = React.useState('any');
+  const [category, setCategorie] = React.useState('any');
   const [skip, setSkip] = React.useState<number>(0);
   const [query, setQuery] = React.useState('');
```
Comment (Bohdan-ya) [inline src/contexts/SearchContext.tsx:30]: @oleksandr-bratashchuk use/or not use generics

File: src/types/models/BasketData.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export interface BasketData {
+  id: number;
```
Comment (Bohdan-ya) [inline src/types/models/BasketData.ts:2]: @oleksandr-bratashchuk check it

**Review by Bohdan-ya — CHANGES_REQUESTED** [review]

**Review by Bohdan-ya — APPROVED** [review]

**Pull request merged**